crates/deltalake-core/src/operations/drop_constraints.rs

```diff
@@ -0,0 +1,247 @@
+    /// Specify if you want to raise if the constraint does not exist
+    pub fn with_raise_if_not_exists(mut self, raise: bool) -> Self {
+        self.raise_if_not_exists = raise;
+        self
+    }
+
+    /// Additional metadata to be added to commit info
+    pub fn with_metadata(
+        mut self,
+        metadata: impl IntoIterator<Item = (String, serde_json::Value)>,
+    ) -> Self {
+        self.app_metadata = Some(HashMap::from_iter(metadata));
+        self
+    }
+}
+
+impl std::future::IntoFuture for DropConstraintBuilder {
+    type Output = DeltaResult<DeltaTable>;
+
+    type IntoFuture = BoxFuture<'static, Self::Output>;
+
+    fn into_future(self) -> Self::IntoFuture {
+        let mut this = self;
+
+        Box::pin(async move {
+            let name = match this.name {
+                Some(v) => v,
+                None => return Err(DeltaTableError::Generic("No name provided".to_string())),
+            };
+
+            let mut metadata = this.snapshot.metadata().clone();
+            let configuration_key = format!("delta.constraints.{}", name);
+
+            let found_constraint = metadata.configuration.contains_key(&configuration_key);
+            if this.raise_if_not_exists && !found_constraint {
+                return Err(DeltaTableError::Generic(format!(
+                    "Constraint with name: {} doesn't exists",
+                    name
+                )));
+            } else if !this.raise_if_not_exists && !found_constraint {
+                return Ok(DeltaTable::new_with_state(this.log_store, this.snapshot));
+            }
+
+            metadata
+                .configuration
+                .remove(format!("delta.constraints.{}", name).as_str());
+
+            let mut contains_constraints: bool = false;
+            for key in metadata.configuration.keys() {
+                if key.starts_with("delta.constraints") {
+                    contains_constraints = true;
+                    break;
+                } else {
+                    continue;
+                }
+            }
+
+            let operational_parameters = HashMap::from_iter([("name".to_string(), json!(&name))]);
+
+            let operations = DeltaOperation::DropConstraint { name: name.clone() };
+
+            let app_metadata = match this.app_metadata {
+                Some(metadata) => metadata,
+                None => HashMap::default(),
+            };
+
+            let commit_info = CommitInfo {
+                timestamp: Some(Utc::now().timestamp_millis()),
+                operation: Some(operations.name().to_string()),
+                operation_parameters: Some(operational_parameters),
+                read_version: Some(this.snapshot.version()),
+                isolation_level: Some(IsolationLevel::Serializable),
+                is_blind_append: Some(false),
+                info: app_metadata,
+                ..Default::default()
+            };
+
+            let protocol = if !contains_constraints {
```

Comment:
Do you know if other implementations downgrade the protocol as a result of dropping the constraint? I can check this for you, but I don't think the protocol says anything about whether or not it's required to do this.

Reply:
I just assumed here it's safe to do since the protocol version was only for constraints, and if there are no constraints imho it should not be left there. 

You could check what spark-delta here does :) But it should be fine I think, I don't believe it violates the protocol

Reply:
Discussed with @roeap on slack and it wouldn't be safe to do this, so I've removed it in ths operation, but we may look at adding an explicit operation to downgrade